### PR TITLE
Move shared snforge collector abstraction to separate crate

### LINKED ISSUE #1216: Crate for structs shared between scarb and snforge
Separate crate that so we don't have to copy structs declarations that we serialize in `scarb-snforge-test-collector` and then deseralize in `forge` (e.g. `forge/src/compiled_raw.rs`)

### PR DESCRIPTION
**Pull request by maciektr**
Closes #1216

Related: https://github.com/software-mansion/scarb/pull/999
## Introduced changes

-

## Checklist

- [ ] Linked relevant issue
- [ ] Updated relevant documentation
- [ ] Added relevant tests
- [ ] Performed self-review of the code
- [ ] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: crates/forge/src/compiled_raw.rs [hunk 1]
```diff
@@ -1,69 +1,10 @@
...
-use starknet::core::types::BlockId;
-use starknet::core::types::BlockTag::Latest;
+use snforge_test_collector_abstractions::{CrateLocation, TestCaseRaw};
 
 #[derive(Debug, Clone, Deserialize)]
 pub(crate) struct CompiledTestCrateRaw {
     pub sierra_program: Program,
     pub test_cases: Vec<TestCaseRaw>,
     pub tests_location: CrateLocation,
 }
```
Comment (piotmag769) [inline crates/forge/src/compiled_raw.rs:10]: This can be moved to the common crate too

File: crates/forge-runner/src/compiled_runnable.rs [hunk 2]
```diff
@@ -26,8 +29,24 @@ pub struct ValidatedForkConfig {
...
+                assert_eq!(self.block_id_value, "Latest");
+                BlockId::Tag(Latest)
+            }
+            _ => unreachable!(),
+        };
+        Ok(ValidatedForkConfig {
+            url: self.url.parse()?,
+            block_id,
+        })
+    }
```
Comment (Arcticae) [inline crates/forge-runner/src/compiled_runnable.rs:51]: Why was this moved here? 🤔

File: crates/forge/src/compiled_raw.rs (hunk 1 above)
Comment (maciektr) [reply crates/forge/src/compiled_raw.rs:10]: No it cannot - it depends on `cairo-lang-sierra`, it would always link the same version in both collector and forge.

File: crates/forge-runner/src/compiled_runnable.rs (hunk 2 above)
Comment (maciektr) [reply crates/forge-runner/src/compiled_runnable.rs:51]: `RawForkParams` has been moved to the shared crate, we cannot implement a trait for it (as neither the trait nor the struct has been defined in this crate)

Comment (piotmag769) [reply crates/forge-runner/src/compiled_runnable.rs:51]: This will be removed with #1049 that I'm working on rn anyways

**Comment by Arcticae** [comment]
Closing since it's heavily outdated
